.github/workflows/ci-build-macos.yaml

```diff
@@ -22,11 +22,9 @@ on:
   pull_request:
     branches:
       - master
-    paths-ignore: ['site2/**', 'deployment/**']
```

Comment:
`paths-ignore` would be fine here, but it's removed to be consistent with other workflows in apache/pulsar where [paths-filter action](https://github.com/apache/pulsar-test-infra/tree/master/paths-filter) (copy of https://github.com/dorny/paths-filter) is used.

> GitHub provides a [paths-ignore](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths) feature out of the box.
> However, GitHub's `paths-ignore` has some limitations:
> 
> - GitHub's `paths-ignore` fails to look at previous commits. This means that the outcome depends on how often you push changes.
> - Consequently, GitHub's `paths-ignore` does not work for [required checks](https://docs.github.com/en/github/administering-a-repository/about-required-status-checks).
>   If you path-ignore a required check, then pull requests will block forever without being mergeable.
> 
([source](https://github.com/marketplace/actions/skip-duplicate-actions#skip-ignored-paths))

`paths-ignore` was once used in apache/pulsar (introduced by #9527), but it had to be reverted in #9606 because of the limitations.